src/util/simplify_expr.cpp

```diff
@@ -1365,17 +1365,12 @@ bool simplify_exprt::simplify_update(exprt &expr)
     {
       const irep_idt &component_name=
         e.get(ID_component_name);
-
-      if(!to_struct_type(value_ptr_type).
-         has_component(component_name))
+      const struct_typet &value_ptr_struct_type=to_struct_type(value_ptr_type);
```

Comment:
The commit message could be a bit more specific regarding the kind of clean-ups this commit is doing.

Reply:
Sorry, that commit is not really a part of this PR and is just included because this depends on some of the changes made in a prior PR